zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditions.java

```diff
@@ -494,6 +502,47 @@ private static void requireNoBoundaryEvent(
     }
   }
 
+  public static void requireMappedBoundaryEventsToStayAttachedToSameElement(
```

Comment:
🔧 This naming explains "what will happen as a result (boundary event will be detached wrongly)". A name that explains what the method does would be a better choice. E.g. `requireMappedBoundaryEventsExistInTarget` which explains what the method does. WDYT?

Reply:
🤔 But `requireMappedBoundaryEventsExistInTarget` is not correct though. It's not testing whether or not the mapped boundary event exists in the target. It checks that the mapped boundary event stays attached to the same element after mapping, i.e. that the boundary event is not detached from the element it was initially attached to.

Do you see any other options?

Reply:
Yeah, I think, the name and the error message you chose is more user friendly. It actually indicates something that cannot be done by user when they try to move the boundary event to a different task. E.g. 
![Screenshot 2024-07-04 at 19 54 51](https://github.com/camunda/camunda/assets/124141078/0177d4ca-e47f-495b-98bb-5b665d39b9df)